### DISCUSSION TITLE
Edit external_url of ModuleItem returns error

### DISCUSSION THREAD

**Discussion by maxjeltes** [ucfopen/canvasapi#633] [Q&A] [answered]
While trying to edit an the external_url of a ModuleItem, I am getting the error:
"BadRequest: 'missing module item parameter'"

I have my API_URL and API_KEY but do not want to show in the code.

```
canvas = Canvas(API_URL, API_KEY)

course = canvas.get_course(13489)
module = course.get_module(65186)
module_item = module.get_module_item(417500)

module_item.edit(new_mi = {'external_url' : 'https://google.com'})
```

I am wondering if anyone knows how to fix this problem. I am following the ucfopen README but I cannot find anything there to help me.

**Comment by Thetwam** [comment, marked as answer]
Hey @maxjeltes,

[According to the Canvas docs for editing a module item](https://canvas.instructure.com/doc/api/modules.html#method.context_module_items_api.update), changing the external URL requires the parameter `module_item[external_url]`. Since brackets aren't allowed in argument names in Python, the CanvasAPI library requires you to use an argument called `module_item` and set it equal to a dictionary containing a key called `external_url`. What you have is close, but the name of the argument is important.

Try the following:

```
module_item.edit(module_item={'external_url' : 'https://google.com'})
```

For more information about how CanvasAPI handles keyword arguments, check out our [keyword argument documentation](https://canvasapi.readthedocs.io/en/stable/keyword-args.html). In particular, take a look at the [Nested Parameters](https://canvasapi.readthedocs.io/en/stable/keyword-args.html#nested-parameters) section.

**Reply by maxjeltes** [reply]
Hi @Thetwam,
Thanks for helping out! The Nested Parameters section in the documentation describes it nicely indeed.
